src/components/CallView/Grid/Grid.vue

```diff
@@ -523,6 +596,34 @@ export default {
 				this.currentPage = Math.max(0, this.numberOfPages - 1)
 			}
 		},
+
+		speakers(speakers) {
+			if (speakers.length === 0) {
+				return
+			}
+			speakers.forEach(speaker => {
+				this.promoteSpeaker(speaker)
+				if (this.unpromoteSpeakerTimer[speaker.attributes.nextcloudSessionId]) {
+					clearTimeout(this.unpromoteSpeakerTimer[speaker.attributes.nextcloudSessionId])
+				}
+			})
+		},
+
+		speakersWithAudioOff(newModels, oldModels) {
+			if (!newModels || newModels.length === 0) {
+				return
+			}
+
+			const addedModels = !oldModels || oldModels.length === 0 ? newModels : newModels.filter(model => !oldModels.includes(model))
+			if (addedModels.length === 0) {
+				return
+			}
+			addedModels.forEach(speaker => {
```

Comment:
And if type is consistent:

```suggestion
			newModels.forEach(speaker => {
				if(!oldModels.includes(speaker)) {
					return
				}
```